SEP/SEP-0007/sep-0007.md

```diff
@@ -179,49 +184,53 @@ the substitution from 18.6 ends up producing
        ))
 ```
 
-The `?x` inside the `SELECT ?y` is not projected out so it is a "different" `?x`
-than the outer one - changing it to another other unused name in the same query
+The `?x` inside the `SELECT ?y` is not projected out, so it is a "different" `?x`
+than the outer one; changing it to another other unused name in the same query
 would not normally affect the query results.
 
-## An Improved "substitute" Operation
-Evalauting
+## An Improved `substitute` Operation
+Evaluating
 [`substitute`](https://www.w3.org/TR/sparql11-query/#defn_substitute) is
-performed for a given solution mapping. For example, the `EXISTS` operation
-evaluates to `true` if a graph pattern has one or more matches given the variable bindings
-of a solution mapping. We call this solution mapping the *current row*
-in this description.
+performed for a given solution mapping. For example, if a graph pattern
+has one or more matches given the variable bindings of a solution mapping,
+the `EXISTS` operation evaluates to `true`. We call this solution mapping
+the *current row* in this description.
 
 This section proposes an alternative mechanism.  Rather than replace each
 variable by the value it is bound to in the current row, this alternative
 mechanism makes the whole of the current row available at any point in the
 evaluation of an `EXISTS` expression. It uses the current row to
-restrict the binding of variables at the points where variable bindings are
-created during evaluation of `EXISTS` to be those from the current row.
+restrict the binding of variables, at the points where variable bindings are
+created during evaluation of `EXISTS`, to be those from the current row.
 It makes illegal syntactic constructs that could lead to an attempt to rebind a
-variable from the current row through using the `AS` syntax.
+variable from the current row through the `AS` syntax.
 
 Section ["Addressing Issues"](#addressing-issues) describes how this alternative
 definition of `substitute` addresses each of the issues identified above.
 
 There are 3 parts to the proposal:
```

Comment:
Note that I left the colon in place here, as it followed a complete sentence. I should have replaced the numeral `3` on the first round, as low numbers (some say single-digit, others say single-word) should be written as words when they occur in prose.
```suggestion
There are three parts to the proposal:
```